### ISSUE TITLE
Enhance the git delta plugin to fetch the test classes to run for the generated delta package

### ISSUE THREAD

**Issue by RohitGagan16** [scolladon/sfdx-git-delta#402]
Hi Team,
 Can we enhance the git delta plugin to fetch the test classes to run for the generated delta package.
 That said, i think this should  cover the dependencies that the classes in the delta package has and also the apex triggers, visual pages tha delta package has. These might call the apex test classes and get us the required coverage of 75% to cover the delta deployment

**Comment by mehdisfdc** [comment]
Hi @RohitGagan16 .
Thank you for logging your enhancement request.

Have you already looked at https://github.com/scolladon/sfdx-git-delta/issues/126 and https://github.com/scolladon/sfdx-git-delta/issues/241? 
Is your request different from has been discussed there?

**Comment by scolladon** [comment]
Hi @RohitGagan16!

Thank you very much for opening this topic.
It is a very interesting topic and a lot of people are interested in.

As @mehdisfdc pointed it, it has been discussed and we provided a simple way to address the point.

My 2 cents here.

**TL;DR**
It's not currently possible, as it would required connection to an org (which SGD does not have today, since it is purely source based).
Orgs can be different based on how the environment are used (and other factors), it means the tool will need to run depending on its target. It means the deployment procedure is different depending on the target which should not be the case. It is best IMHO to not break the repeatability of a deployment process.
It requires complex advanced algorithm based on data extracted from an org to compute minimal test classes to run.
If it should be done (I think not), I definitely think it should be done by another dedicated plugin around apex test.

**Requirements:**
The requirements are:
- to create a list of apex test classes, from a package.xml or a source folder
- to ensure minimal test execution to cover at least 75% of the package deployed.

**Solutions**
The solution should also be able to cover Flow as well.
This requirement can be split in 3 maturity steps
- **_(crawl maturity 👶)_** using a Naming convention and build the list of apex test class from the package.xml
If your project is using a naming convention to define test classes related to a class and you have a guideline of ensuring this test class covers its class above 75%, then it's just a matter of building the list of test class from the package.xml.
We already cover [how to deal with it](https://github.com/scolladon/sfdx-git-delta#generate-a-comma-separated-list-of-the-added-and-modified-apex-classes) and we also [provided an example](https://github.com/scolladon/sfdx-git-delta/issues/241#issuecomment-1023033304) of doing it with a file name pattern
- **_(walk maturity 🚶)_** using the Dependency API or the Tooling Symbol table it seems achievable to list the test classes using the public API of the class we want to test. In this situation, no need to have a naming convention, it can deduce which test class cover which class. it requires to have a connection to the org (which sgd does not requires). It requires advanced algorithm for something that should not be a problem when unit testing is done properly IMHO
- **_(run maturity 🏃 )_** using the Apex Coverage Table as well as the dependency api it seems to be possible to deduce the minimal tests to run in order to cover the classes. it requires to have a connection to the org (which sgd does not requires). It requires the data to be in the org, meaning the coverage already ran fully for the content of the package we are trying to deploy. It requires more advanced algorithm for something that should not be a problem when unit testing is done properly IMHO

**My opinion**
SGD does not need connection to an org, it is source based.
You can recreate any package created previously thanks to git (idempotence and repeatability).
IMHO I think we should not need to compute the shortest path of test classes to execute in order to cover classes when [proper Software Unit Test technique](https://www.youtube.com/watch?v=hMfJXAhHXcU) are in place, because sane  testing strategy should cover any classe above 80% mechanically. I would advocate for that.
I would really advocate against doing it in SGD. And against doing it at all because it would move us away.
